Upgrade meilisearch-java to v0.20.1

### Description

The project currently uses `com.meilisearch.sdk:meilisearch-java:0.15.0`.

Maven reports that `0.20.1` is available:

```
com.meilisearch.sdk:meilisearch-java ................ 0.15.0 -> 0.20.1
```

A first verification with the upgraded SDK compiles, but tests fail at runtime because OkHttp classes are missing:

```
./mvnw test -Dmeilisearch-java=0.20.1

Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.meilisearch.sdk.http.CustomOkHttpClient
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NoClassDefFoundError: okhttp3/MediaType
```

### Notes

`meilisearch-java:0.20.1` pulls `com.squareup.okhttp3:okhttp:5.3.2`. The JVM class appears to be provided by `com.squareup.okhttp3:okhttp-jvm:5.3.2`, so the upgrade may require explicitly adding or aligning the OkHttp JVM dependency.

This is related to the previous OkHttp class loading issue tracked in #168.

### Acceptance Criteria

- Upgrade `meilisearch-java` to `0.20.1` or the latest compatible release.
- Resolve the OkHttp `NoClassDefFoundError` during Spring context initialization.
- Verify with `./mvnw test`.

--- a/pom.xml
+++ b/pom.xml
@@ -26,7 +26,8 @@
 
     <properties>
         <springdata.commons>3.3.13-SNAPSHOT</springdata.commons>
-        <meilisearch-java>0.15.0</meilisearch-java>
+        <meilisearch-java>0.20.1</meilisearch-java>
+        <okhttp>5.3.2</okhttp>
         <testcontainers-meilisearch>1.0.5</testcontainers-meilisearch>
 
         <java-module-name>spring.data.meilisearch</java-module-name>
@@ -64,6 +65,11 @@
             <groupId>com.meilisearch.sdk</groupId>
             <artifactId>meilisearch-java</artifactId>
             <version>${meilisearch-java}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.squareup.okhttp3</groupId>
+            <artifactId>okhttp-jvm</artifactId>
+            <version>${okhttp}</version>
         </dependency>
 
         <!-- Jackson JSON Mapper -->

